apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/SizedBufferingConsumer.cs

```diff
@@ -23,8 +25,8 @@ namespace Google.Cloud.Diagnostics.Common
     /// </summary>
     internal class SizedBufferingConsumer<T> : IFlushableConsumer<T>
     {
-        /// <summary>A mutex to protect the buffer.</summary>
-        private readonly object _mutex = new object();
+        /// <summary>A semaphore to protect the buffer.</summary>
+        private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1);
```

Comment:
Ah, presumably it's so you can unlock it from a different thread after an await?